python/pyarrow/table.pxi

```diff
@@ -536,14 +548,16 @@ cdef class RecordBatch:
         return OrderedDict(entries)
 
 
-    def to_pandas(self, nthreads=None):
+    def to_pandas(self, nthreads=None, str_to_categorical=False):
```

Comment:
This option needs to be pushed down into `ConvertTableToPandas` in https://github.com/apache/arrow/blob/master/cpp/src/arrow/python/arrow_to_pandas.h#L50. I even have a "PandasOptions" struct with `strings_to_categorical` waiting for you =)